### PR TITLE
feat(metrics): add ability to use OTLP source

### PR DESCRIPTION
**Pull request by swiatekm-sumo**
Add a setting for switching to the OTLP source for metrics.

The source is created for all users with setup enabled, and the setting only controls if we're sending data to it. It's intentionally one source vs the 8 with have for HTTP, as I believe the technical reasons for the latter are no longer valid.

In terms of implementation, parts of the metrics pipeline become conditional.

### Checklist

- [x] Changelog updated or skip changelog label added
- [x] Documentation updated
- [x] Template tests added for new features
- [x] Integration tests added or modified for major features

### CODE REVIEW

File: deploy/helm/sumologic/conf/metrics/otelcol/config.yaml [hunk 1]
```diff
@@ -333,14 +337,18 @@ service:
...
         - sumologic/default
+{{- if eq .Values.sumologic.metrics.sourceType "http" }}
         - sumologic/apiserver
         - sumologic/control_plane
         - sumologic/controller
         - sumologic/kubelet
         - sumologic/node
         - sumologic/scheduler
         - sumologic/state
+{{- end }}
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/conf/metrics/otelcol/config.yaml:353]: Will that work fine (without any blacklisting and throttling due to metrics cardinality)?

Comment (swiatekm-sumo) [reply deploy/helm/sumologic/conf/metrics/otelcol/config.yaml:353]: In the meantime, can you review these changes on the assumption that this ok to do?

Comment (sumo-drosiek) [reply deploy/helm/sumologic/conf/metrics/otelcol/config.yaml:353]: Yes, remaining changes looks fine for me

File: deploy/helm/sumologic/values.yaml [hunk 2]
```diff
@@ -164,6 +164,11 @@ sumologic:
         default:
           name: (default-metrics)
           config-name: endpoint-metrics
+        default-otlp:
+          name: default-metrics-otlp
```
Comment (astencel-sumo) [inline deploy/helm/sumologic/values.yaml:168]: Can we standardize the naming? The OTLP source for logs is named `logs-otlp`, can we name this simply `metrics-otlp`?

Comment (swiatekm-sumo) [reply deploy/helm/sumologic/values.yaml:168]: Sure, will do.

File: deploy/helm/sumologic/conf/metrics/otelcol/exporters.yaml [hunk 3]
```diff
@@ -1,106 +1,112 @@
...
-  endpoint: ${SUMO_ENDPOINT_APISERVER_METRICS_SOURCE}
-  max_request_body_size: 16_777_216  # 16 MB before compression
-  metric_format: prometheus
+## Configuration for Sumo Logic Exporter
+## ref: https://github.com/SumoLogic/sumologic-otel-collector/blob/main/pkg/exporter/sumologicexporter
+sumologic/default:
+  metric_format: {{ include "metrics.otelcol.exporter.format" . }}
+  endpoint: {{ include "metrics.otelcol.exporter.endpoint" . }}
+  ## Configuration for sending queue
+  ## ref: https://github.com/open-telemetry/opentelemetry-collector/tree/release/v0.37.x/exporter/exporterhelper#configuration
```
[the remark this answers is not in the document]
Comment (swiatekm-sumo) [reply deploy/helm/sumologic/conf/metrics/otelcol/exporters.yaml:7]: The exporters? Or keys inside the exporter? Or both? :dagger:

[the remark this answers is not in the document]
Comment (sumo-drosiek) [reply deploy/helm/sumologic/conf/metrics/otelcol/exporters.yaml:7]: its fine for me especially we dont have strict rule on that yet

**Review by sumo-drosiek — APPROVED** [review]

**Pull request merged**